pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaAvroMessageDecoder.java

```diff
@@ -111,7 +110,7 @@ public GenericRow decode(byte[] payload, int offset, int length, GenericRow dest
               length - HEADER_LENGTH, null));
       return avroRecordConvetrer.transform(avroRecord, schema, destination);
     } catch (IOException e) {
-      LOGGER.error("Caught exception while reading message", e);
+      LOGGER.error("Caught exception while reading message using schema {}", schema.getName(), e);
```

Comment:
It doesn't look like it's possible for `schema` to be `null` at this point in time, but can you guard against it in case it happens in a code change/refactoring?